File: app/src/main/java/com/cn29/aac/ui/shopping/AlbumFragment.java
package com.cn29.aac.ui.shopping;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;
import com.cn29.aac.databinding.FragmentAlbumBinding;
import com.cn29.aac.di.scope.AndroidDataBinding;
import com.cn29.aac.di.scope.ViewModel;
import com.cn29.aac.repo.itunes.Album;
import com.cn29.aac.ui.base.BaseFragment;
import com.cn29.aac.ui.common.BaseRecyclerViewAdapter.OnItemClickListener;
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModel;
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel;
import java.util.List;
import javax.inject.Inject;

public class AlbumFragment extends BaseFragment {

  @Inject
  @ViewModel
  AlbumFragmentViewModel viewModel;

  @Inject
  @ViewModel
  ShoppingActivityViewModel shoppingActivityViewModel;

  @Inject
  @AndroidDataBinding
  FragmentAlbumBinding binding;


  private AlbumAdapter adapter;

  private List<Album> albums;

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    int artistId = 909253; //default value
    String entity = "album"; //default value
    if (getArguments() != null && getArguments().containsKey("artistId") && getArguments()
        .containsKey("entity")) {
      artistId = getArguments().getInt("artistId");
      entity = getArguments().getString("entity");
    }
    viewModel.getAlbumResult(artistId, entity).observe(this, response ->
    {
      switch (response.status) {
        case CHECKING:
          break;
        case SUCCESS:
          progressDialogComponent.hideLoading();
          albums = response.data;
          adapter = new AlbumAdapter(albums,
              getAlbumOnItemClickListener());
          binding.list.setAdapter(adapter);
          break;
        case ERROR:
          progressDialogComponent.hideLoading();
          break;
        case LOADING:
          progressDialogComponent.showLoading();
          break;
      }
    });
  }

  @NonNull
  private OnItemClickListener<Album> getAlbumOnItemClickListener() {
    return item -> {
      //add to the kart
      Toast.makeText(getActivity(), "Implement your own", Toast.LENGTH_SHORT).show();
    };
  }


  @Override
  public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    // Inflate the layout for this fragment
    return binding.getRoot();
  }

}

File: app/src/main/java/com/cn29/aac/ui/shopping/AlbumFragment.kt
package com.cn29.aac.ui.shopping

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.Observer
import com.cn29.aac.databinding.FragmentAlbumBinding
import com.cn29.aac.di.scope.AndroidDataBinding
import com.cn29.aac.di.scope.ViewModel
import com.cn29.aac.repo.itunes.Album
import com.cn29.aac.repo.util.Resource
import com.cn29.aac.repo.util.Status
import com.cn29.aac.ui.base.BaseFragment
import com.cn29.aac.ui.common.BaseRecyclerViewAdapter
import com.cn29.aac.ui.shopping.vm.AlbumFragmentViewModel
import com.cn29.aac.ui.shopping.vm.ShoppingActivityViewModel
import javax.inject.Inject

class AlbumFragment : BaseFragment() {
    @Inject
    @ViewModel
    lateinit var viewModel: AlbumFragmentViewModel

    @Inject
    @ViewModel
    lateinit var shoppingActivityViewModel: ShoppingActivityViewModel

    @Inject
    @AndroidDataBinding
    lateinit var binding: FragmentAlbumBinding
    private var adapter: AlbumAdapter? = null
    private var albums: List<Album>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var artistId = 909253 //default value
        var entity: String? = "album" //default value
        if (arguments?.containsKey("artistId") == true && arguments?.containsKey(
                        "entity") == true) {
            artistId = arguments!!.getInt("artistId")
            entity = arguments!!.getString("entity")
        }
        viewModel
                .getAlbumResult(artistId, entity)
                .observe(this,
                         Observer { response: Resource<List<Album>> ->
                             when (response.status) {
                                 Status.CHECKING -> {
                                 }
                                 Status.SUCCESS -> {
                                     progressDialogComponent!!.hideLoading()
                                     albums = response.data
                                     adapter = AlbumAdapter(
                                             albums!!,
                                             albumOnItemClickListener)
                                     binding.list.adapter = adapter
                                 }
                                 Status.ERROR -> progressDialogComponent!!.hideLoading()
                                 Status.LOADING -> progressDialogComponent!!.showLoading()
                             }
                         })
    }

    //add to the kart
    private val albumOnItemClickListener: BaseRecyclerViewAdapter.OnItemClickListener<Album>
        get() = object : BaseRecyclerViewAdapter.OnItemClickListener<Album> {
            override fun onItemClick(item: Album) {
                //add to the kart
                Toast.makeText(
                                activity,
                                "Implement your own",
                                Toast.LENGTH_SHORT)
                        .show()
            }
        }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return binding.root
    }
}